### PR TITLE
[AIRFLOW-807] Improve scheduler performance for large DAG Python file…

### PR DESCRIPTION
**Pull request by criccomini**
…s with MySQL

Dear Airflow Maintainers,

Please accept this PR that addresses the following issues:
- *(replace with a link to AIRFLOW-807)*

Testing Done:
- Running in production. Seeing 15x-20x performance increase.

### CODE REVIEW

**Comment by artwr** [comment]
+1

File: airflow/jobs.py [hunk 1]
```diff
@@ -538,12 +538,17 @@ def manage_slas(self, dag, session=None):
         Where assuming that the scheduler runs often, so we only check for
         tasks that should have succeeded in the past hour.
         """
+        if not any([ti.sla for ti in dag.tasks]):
+            self.logger.info('Skipping SLA check for {} because no tasks in DAG have SLAs'.format(dag))
```
Comment (bolkedebruin) [inline airflow/jobs.py]: Can you fix the landscape issue here (column count)

File: airflow/jobs.py [hunk 2]
```diff
@@ -538,12 +538,17 @@ def manage_slas(self, dag, session=None):
...
+            return
+
         TI = models.TaskInstance
         sq = (
             session
             .query(
                 TI.task_id,
                 func.max(TI.execution_date).label('max_ti'))
+            .with_hint(TI, 'USE INDEX (PRIMARY)', dialect_name='mysql')
```
Comment (bolkedebruin) [inline airflow/jobs.py:552]: Acceptable for now: issue seems to be more prevalent as primary almost never seems to be used. The query optimiser is really weird here.

File: airflow/jobs.py [hunk 3]
```diff
@@ -538,12 +538,18 @@ def manage_slas(self, dag, session=None):
         Where assuming that the scheduler runs often, so we only check for
         tasks that should have succeeded in the past hour.
         """
+        if not any([ti.sla for ti in dag.tasks]):
+            self.logger.info('Skipping SLA check for {} because ' + \
```
Comment (bolkedebruin) [inline airflow/jobs.py]: ehm really a nit, but double quotes would make this look a bit cleaner?

**Pull request closed without merging**